pkg/machine/pull.go

```diff
@@ -91,6 +93,19 @@ func (g GenericDownload) HasUsableCache() (bool, error) {
 	return g.URL == nil, nil
 }
 
+// CleanCache cleans out downloaded uncompressed image files
+func (g GenericDownload) CleanCache() error {
+	// Remove any image that has been downloaded via URL
+	// We never read from cache for generic downloads
+	if g.URL != nil {
```

Comment:
before we prune, should we make sure that the file is not being used by a different machine?

Reply:
I don't think that should matter, since we always make a copy of the the image for each VM. Since we never check the cache for this image, we will pull this image always, even if it has been downloaded before.